compiler/src/dotty/tools/dotc/core/Symbols.scala

```diff
@@ -265,12 +266,32 @@ object Symbols {
 
     /** The source or class file from which this class or
      *  the class containing this symbol was generated, null if not applicable.
-     *  Note that this the returned classfile might be the top-level class
+     *  Note that the returned classfile might be from the top-level class
      *  containing this symbol instead of the directly enclosing class.
-     *  Overridden in ClassSymbol
      */
     def associatedFile(using Context): AbstractFile | Null =
-      lastDenot.topLevelClass.associatedFile
+      val compUnitInfo = compilationUnitInfo
+      if compUnitInfo == null then (null: AbstractFile | Null)
+      else compUnitInfo.associatedFile
+
+    /** The compilation unit info (associated file, tasty versions, ...).
+     *  Note that the returned CompilationUnitInfo might be from the top-level class
+     *  containing this symbol instead of the directly enclosing class.
+     *  Overridden in ClassSymbol
+     */
+    def compilationUnitInfo(using Context): CompilationUnitInfo | Null =
+      lastDenot.topLevelClass.compilationUnitInfo
+
+    /** The version of TASTy from which the symbol was loaded, None if not applicable. */
+    def tastyVersion(using Context): Option[TastyVersion] =
+      val compUnitInfo = compilationUnitInfo
+      if compUnitInfo == null then None
+      else compUnitInfo.tastyVersion
+
+    /** If this class has explicit nulls enabled */
+    def explicitNulls(using Context): Boolean =
```

Comment:
@noti0na1 here is what you need to know if a symbol (from TASTy or source) was compiled using explicit nulls.

Reply:
Thanks!